src/V3Randomize.cpp

```diff
@@ -1331,6 +1331,24 @@ class RandomizeVisitor final : public VNVisitor {
             }
             AstMemberDType* const firstMemberp = unionDtp->membersp();
             return newRandStmtsp(fl, exprp, nullptr, offset, firstMemberp);
+        } else if (AstUnpackArrayDType* const unpackarrayDType = VN_CAST(
+                       memberp ? memberp->subDTypep()->skipRefp() : exprp->dtypep()->skipRefp(),
+                       UnpackArrayDType)) {
+            AstNodeStmt* stmtsp = nullptr;
+            int elemCount = unpackarrayDType->elementsConst();
+            for (int i = 0; i < elemCount; ++i) {
```

Comment:
This can generate lots of code for large arrays, so a loop might be better (smaller resulting code, faster compilation etc). I would personally go with AstForeach (which can now be used thanks to #5283), so that we can do all array types the same way.

Reply:
The stmtsp here should probably be just the result of a recursive newRandStmtsp call. But they can be any statement node that will be executed for every value of the index. See #5448 for my approach to creating AstForeach (some parts of the PR are specific to putting a loop of statements inside of an expression, but you can safely ignore all AstText manipulation, none of that should be needed here).